src/app/pages/settings-page/settings-page.component.ts

```diff
@@ -0,0 +1,54 @@
+import { CommonModule } from '@angular/common';
+import { CUSTOM_ELEMENTS_SCHEMA, Component, OnInit } from '@angular/core';
+import { AbstractControl, FormControl, FormGroup, ReactiveFormsModule, ValidationErrors, Validators } from '@angular/forms';
+import { UserStoreService } from 'src/app/services/user-store.service';
+
+@Component({
+  selector: 'app-settings-page',
+  standalone: true,
+  templateUrl: './settings-page.component.html',
+  styleUrls: ['./settings-page.component.css'],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
+  imports: [ReactiveFormsModule,CommonModule],
+})
+export default class SettingsPageComponent implements OnInit {
+  constructor(private userStoreService: UserStoreService) {}
+
+  userName = '';
+  userNameError = false;
+  placeOfWork = '';
+  placeOfWorkError = false;
+
+ 
+  
+  formField: FormGroup = new FormGroup({
+    userName: new FormControl('', Validators.minLength(10)),
+    placeOfWork: new FormControl('', Validators.required),
+  });
+
+
+  ngOnInit() {
+    this.userStoreService.userName$.subscribe((userName) => {
+      this.userName = userName;
+    });
+
+    this.userStoreService.placeOfWork$.subscribe((placeOfWork) => {
+      this.placeOfWork = placeOfWork;
+    });
```

Comment:
Every time an observable is subscribed to "manually" we have to handle the unsubscription for that observable.

a common pattern is for example: 

1. declare a subject on the component level
2. use the pipe before the subscribe & the takeUntil operator with the subject
3.  handle the unsubscribe in the destroy method

example:

```
export default class SettingsPageComponent implements OnInit, OnDestroy {
  constructor(private userStoreService: UserStoreService) {}
  private onDestroy$: Subject<void> = new Subject<void>()
  
  ngOnInit() {
    this.userStoreService.userName$
      .pipe(
        takeUntil(this.onDestroy$)
      )
      .subscribe((userName) => {
        this.userName = userName;
      });
  }

  ngOnDestroy(): void {
    this.onDestroy$.next()
    this.onDestroy$.complete()
  }
}

```

Reply:
the pipe with takeUntil can be added to multiple different subscriptions in the same compoennt

Reply:
An alternative approach , that I created for re-usability. (thanks to angular 16, destroyREF)

They are doing the same job tho :) 

https://github.com/scania-digital-design-system/tegel-angular-demo/pull/26/commits/cc417c35806755c5df5dacc59a88113487353b91

@Tommy-Hallgren

Reply:
i think that is experimental in v.16 and will be released in 17

Reply:
import { takeUntilDestroyed } from '@angular/core/rxjs-interop';